### ISSUE TITLE
Overloading regression in Scala 3 (with implicit conversion)

### ISSUE THREAD

**Issue by unkarjedy** [lampepfl/dotty#15001]
similar to https://github.com/lampepfl/dotty/issues/14699

The following works in 2.13, but fails in Scala 3.

## Compiler version
3.1.2

## Minimized code

```Scala
import scala.language.implicitConversions

trait StringShouldWrapperForVerb {
  def should(right: String): Unit = ???
  def should(right: => Unit): Unit = ???
}

object Example {
  implicit def convertToStringShouldWrapperForVerb(o: String): StringShouldWrapperForVerb = ???

  //ERROR
  "test 1" should {
  }
  
  //OK
  "test 2" should {
    ()
  }
  
  val s: StringShouldWrapperForVerb = ???
  s should {
    ()
  }

  s should {
  }
}
```

[Scala 3 scastie](https://scastie.scala-lang.org/NbNOLFw0RoqjuS9SLhHsqQ)
[Scala 2 scastie](https://scastie.scala-lang.org/mCZn5qgZSTmD46CeL315ug)

## Output
```
None of the overloaded alternatives of method should in trait StringShouldWrapperForVerb with types
 (right: => Unit): Unit
 (right: String): Unit
match arguments (<nonsensical><notype></nonsensical>)
```

## Expectation
No error, same as in Scala 2.13?

**Comment by unkarjedy** [comment]
I accidently noticed this issue when upgrading tests for ScalaTest integration in IntelliJ to Scala 3 & ScalaTest 3.2.11.
(not that it's critical to me)

Real example
```scala
lazy val scalatest_3_2_11 = project.settings(
  scalaVersion := "3.1.2",
  libraryDependencies ++= Seq(
    "org.scalatest" %% "scalatest" % "3.2.11"
  )
)
```

```scala
import org.scalatest._
import org.scalatest.wordspec._

class WordSpecTest extends AnyWordSpec {
  "empty 1" should {
  }

  "empty 2" should {
    "ignore other tests" in {
      print(">>TEST: FAILED <<")
    }
  }
}
```

![image](https://user-images.githubusercontent.com/3989292/164511093-53b678e5-64ed-470f-b5c0-488344fc9f52.png)

**Comment by smarter** [comment]
This works on 3.1.3-RC2 so I guess it was fixed at the same time as https://github.com/lampepfl/dotty/issues/14699

**Comment by unkarjedy** [comment]
Indeed, thanks!

I thought that it was fixed in 3.1.2, cause I saw no error in the example from #14699 
But turns out I just used the same file, and it showed the first error, hiding the error from the ticket =D

![image](https://user-images.githubusercontent.com/3989292/164512806-5879e074-882e-46b3-aca7-4461b56d8d56.png)


![image](https://user-images.githubusercontent.com/3989292/164512657-4976917a-e468-4ee0-887e-792f4406c926.png)
